ee/agent/reset.go

```diff
@@ -112,6 +105,49 @@ func DetectAndRemediateHardwareChange(ctx context.Context, k types.Knapsack) {
 	}
 }
 
+func GetResetRecords(ctx context.Context, k types.Knapsack) ([]dbResetRecord, error) {
+	resetRecords := make([]dbResetRecord, 0)
+	if k.PersistentHostDataStore() == nil {
+		return resetRecords, UninitializedStorageError{}
+	}
+
+	resetRecordsRaw, err := k.PersistentHostDataStore().Get(hostDataKeyResetRecords)
+	if err != nil {
+		return resetRecords, err
+	}
+
+	if len(resetRecordsRaw) == 0 {
+		return resetRecords, nil
+	}
+
+	if err := json.Unmarshal(resetRecordsRaw, &resetRecords); err != nil {
+		return resetRecords, err
+	}
+
+	return resetRecords, nil
+}
+
+func ResetDatabase(ctx context.Context, k types.Knapsack, resetReason string) error {
+	backup, err := prepareDatabaseResetRecords(ctx, k, resetReason)
+	if err != nil {
+		k.Slogger().Log(ctx, slog.LevelWarn, "could not prepare db reset records", "err", err)
```

Comment:
```suggestion
		k.Slogger().Log(ctx, slog.LevelError, "could not prepare db reset records", "err", err)
```

Reply:
Wow. This is an interesting question. Should the reset function continue or abort in the face of errors. I'm good either way, but we may need to revisit.